OSDOCS-2272, changes to STS installation (#34960)
openshift/openshift-docs: modules/rosa-sts-creating-cluster.adoc

.Procedure

. Create the required account-wide roles and policies, including Operator policies:
+
[source,terminal]
----
$ rosa create account-roles --version 4.8 <1>
----
<1> You must specify an OpenShift release by using the `--version` option.
+
You can select from the following modes in the interactive prompt:
+
* `auto`: The account roles and and policies are created directly using the current AWS account.

> * `manual`: The role and policy JSON files are saved in the current directory. The command output includes the `aws` CLI commands required to create the roles.

Reviewer: Should we clarify the aws-cli version that used for the manual mode commands execution? I used 2.0.x aws-cli before which cannot create the policy with option --tags in the output command. Then it worked after I upgrade the aws-cli to 2.4.x

Reviewer 2: Thank you for highlighting this. Do you know if 2.4 is the minimum required version for this functionality?

Reviewer 2: I have reviewed the `aws-cli` changelog and cannot see a reference to this. As discussed, I have subsequently updated the prerequisite to "* You have installed and configured the latest AWS, ROSA, and `oc` CLIs on your installation host."

+
[NOTE]
====
Alternatively, you can specify the mode by using the `--mode auto -y` or `--mode manual` CLI arguments.
====

. Create a cluster by specifying the custom settings shown below or by using the interactive mode. To view other options when creating a cluster, enter `rosa create cluster --help`.
+
Creating a cluster can take up to 40 minutes.

> +

Reviewer 3: This block (and the `[NOTE]` below) belongs to the `rosa create cluster` command.

Reviewer 2: @vkareh thank you for your further review! Can you please expand on this comment? The block that this relates to is part of the step that provides options for the create cluster commands. In the preview, the section includes the above text and then two sub-bullet points that provide the custom command and interactive command options for `rosa cluster create`. Would you like it to be positioned a different way? Thanks.

Reviewer 3: Okay I see it now, my bad. Looking at the adoc code, it seemed as though it was in the previous section, but looking at the preview everything looks in the right place :+1:

Reviewer 2: Great, thank you for confirming that.

>   --support-role-arn arn:aws:iam::${aws_account_id}:role/ManagedOpenShift-Support-Role \

Reviewer: The role arns here depends on the output of last command `$ rosa create account-roles` once user input different prefix rather than `ManagedOpenShift` , then the command here will misleading users' creation

Reviewer 2: Thank you for highlighting this. I have added the prefix to the `rosa create account-roles --version 4.8` command and have added call outs describing what is required if this is customised.

>   --support-role-arn arn:aws:iam::${aws_account_id}:role/ManagedOpenShift-Support-Role \

Reviewer: The role-arn created by account-roles should be `--role-arn arn:aws:iam::301721915996:role/ManagedOpenShift-Installer-Role` not `--role-arn arn:aws:iam::${aws_account_id}:role/ManagedOpenShift-IAM-Role`

Reviewer 2: Thank you for highlighting this. I have updated the example in the PR.

I: Once the cluster is 'Ready' you will need to add an Identity Provider and define the list of cluster administrators. See `rosa create idp --help` and `rosa create user --help` for more information.
I: To determine when your cluster is Ready, run `rosa describe cluster <cluster_name>`.
----
+
[NOTE]
====
You can configure the following default network IP ranges:

* Machine CIDR: 10.0.0.0/16
* Service CIDR: 172.30.0.0/16
* Pod CIDR: 10.128.0.0/14

For the CIDR-related `rosa` CLI arguments, see `rosa create cluster --help | grep cidr`. In the interactive mode, you are prompted for the settings.
====

* To create a cluster with STS using interactive prompts:
+
[source,terminal]
----
$ rosa create cluster --interactive
----

. Create the cluster-specific Operator IAM roles. The roles created include the relevant prefix for the cluster name:
+
[source,terminal]
----
$ rosa create operator-roles --cluster <cluster_name|cluster_id> <1>
----
<1> Replace `<cluster_name|cluster_id>` with the cluster name or the ID of the cluster.
+
You can select from the following modes in the interactive prompt:
+

> * `auto`: The account roles and and policies are created directly using the current AWS account.

Reviewer 3: ```suggestion
* `auto`: The operator roles are created directly using the current AWS account.
```

Reviewer 2: Thank you for highlighting this. I have now resolved this suggestion.

> * `manual`: The role and policy JSON files are saved in the current directory. The command output includes the `aws` CLI commands required to create the roles.

Reviewer 3: ```suggestion
* `manual`: The role JSON files are saved in the current directory. The command output includes the `aws` CLI commands required to create the roles.
```

Reviewer 2: Resolved.

+
[NOTE]
====
Alternatively, you can specify the mode by using the `--mode auto -y` or `--mode manual` CLI arguments.
====

. Create the OpenID Connect (OIDC) provider that the Operators will use to authenticate:
+
[source,terminal]
----

> $ rosa create oidc-provider --mode manual --cluster <cluster_name|cluster_id> <1>

Reviewer: Previous command examples remove --mode to make user go to interactive mode. How about this one? I guess it's better to follow same rule.

Reviewer 2: Good catch, thank you! I will remove this.

----
<1> Replace `<cluster_name|cluster_id>` with the cluster name or the ID of the cluster.
+
You can select from the following modes in the interactive prompt:
+
* `auto`: The OIDC provider is created directly using the current AWS account.

> * `manual`: The OIDC provider JSON files and the thumbprint are saved in the current directory. The command output includes the `aws` CLI commands required to create the OIDC provider.

Reviewer 3: ```suggestion
* `manual`: The command output includes the `aws` CLI commands required to create the OIDC provider.
```
(there are no files generated by this command)

Reviewer 2: Thank you for clarifying this. Resolved.
